custom_components/tahoma/coordinator.py

```diff
@@ -114,11 +118,18 @@ async def _async_update_data(self) -> Dict[str, Device]:
                 EventName.DEVICE_CREATED,
                 EventName.DEVICE_UPDATED,
             ]:
-                self.devices = await self._get_devices()
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(self._config_entry_id)
+                )
+                return None
 
             elif event.name == EventName.DEVICE_REMOVED:
+                base_device_url, *_ = event.deviceurl.split("#")
                 registry = await device_registry.async_get_registry(self.hass)
-                registry.async_remove_device(event.deviceurl)
+
+                if device := registry.async_get_device({(DOMAIN, base_device_url)}):
```

Comment:
1st time I see someone use this :)

Reply:
I took it from here https://github.com/custom-components/integration_blueprint/pull/60/files :D

Reply:
I took it from uptimerobot in core! https://github.com/home-assistant/core/blob/8103d9ae3c12664cdec27d742481cca28eee0796/homeassistant/components/uptimerobot/__init__.py#L79